internal/pkg/manifest/validate.go

```diff
@@ -734,3 +813,110 @@ func (d *DeadLetterQueue) Validate() error {
 func (*OverrideRule) Validate() error {
 	return nil
 }
+
+type validateDependenciesOpts struct {
+	mainContainerName string
+	sidecarConfig     map[string]*SidecarConfig
+	imageConfig       *Image
+}
+
+func validateContainerDeps(opts validateDependenciesOpts) error {
+	if err := validateImageDependsOnStatus(opts); err != nil {
+		return err
+	}
+	if err := validateSidecarsDependsOnStatus(opts); err != nil {
+		return err
+	}
+	return validateNoCircularDependencies(opts)
+}
+
+func validateImageDependsOnStatus(opts validateDependenciesOpts) error {
+	if opts.imageConfig == nil || opts.imageConfig.DependsOn == nil {
+		return nil
+	}
+	for name, status := range opts.imageConfig.DependsOn {
+		if !isEssentialContainer(name, opts) {
+			continue
+		}
+		if err := isValidEssentialStatus(name, strings.ToUpper(status)); err != nil {
+			return fmt.Errorf("validate %s container dependencies status: %w", opts.mainContainerName, err)
+		}
+	}
+	return nil
+}
+
+func validateSidecarsDependsOnStatus(opts validateDependenciesOpts) error {
+	if opts.sidecarConfig == nil {
+		return nil
+	}
+	for s, config := range opts.sidecarConfig {
+		if config.DependsOn == nil {
+			return nil
+		}
+		for name, status := range config.DependsOn {
+			if !isEssentialContainer(name, opts) {
+				continue
+			}
+			if err := isValidEssentialStatus(name, strings.ToUpper(status)); err != nil {
+				return fmt.Errorf("validate %s container dependencies status: %w", s, err)
+			}
+		}
+	}
+	return nil
+}
+
+func isEssentialContainer(name string, opts validateDependenciesOpts) bool {
+	if name == opts.mainContainerName {
+		return true
+	}
+	if opts.sidecarConfig == nil {
+		return false
+	}
+	return opts.sidecarConfig[name].Essential == nil || aws.BoolValue(opts.sidecarConfig[name].Essential)
+}
+
+func isValidEssentialStatus(name, status string) error {
+	for _, allowed := range essentialContainerValidStatuses {
+		if status == allowed {
+			return nil
+		}
+	}
+	return fmt.Errorf("essential container %s can only have status %s", name, english.WordSeries([]string{dependsOnStart, dependsOnHealthy}, "or"))
+}
+
+func validateNoCircularDependencies(opts validateDependenciesOpts) error {
+	dependencies, err := buildDependencyGraph(opts)
+	if err != nil {
+		return err
+	}
+	cycle, acyclic := dependencies.IsAcyclic()
```

Comment:
tiny nit: maybe `acyclic` could be renamed as `ok` since given the function name `IsAcyclic`, it's clear that `ok` means it is indeed acyclic.